azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -334,7 +348,30 @@ private void populateJobTypeForFlow(final ExecutableNode node, Set<String> jobTy
    * @throws ExecutorManagerException
    */
   private void createService(final int executionId) throws ExecutorManagerException {
-    // TODO: Add integration with Service spec builder once that code is available in master branch
+    final V1Service serviceObject;
+    try {
+      final AzKubernetesV1ServiceBuilder azKubernetesV1ServiceBuilder =
+          new AzKubernetesV1ServiceBuilder(
+              "v1Service.yaml");
+      serviceObject = azKubernetesV1ServiceBuilder
+          .withExecId(String.valueOf(executionId))
+          .withServiceName(getServiceName(executionId))
+          .withNamespace(this.namespace)
+          .withApiVersion(SERVICE_API_VERSION_2)
+          .withKind(MAPPING)
+          .withPort(String.valueOf(this.servicePort))
+          .withTimeoutMs(String.valueOf(this.serviceTimeout))
+          .build();
+    } catch (final IOException e) {
+      logger.error("Unable to create service object as service spec is missing: " + e.getMessage());
```

Comment:
Nit:
The error message seems quite specific to the working of AzKubernetesV1ServiceBuilder  for IOException ?
Maybe keep the error message generic or alternatively convert the IOException to something more generic within  AzKubernetesV1ServiceBuilder.build() ?

Reply:
Good point. Fixed it.